spacemesh/v2alpha1/tx.proto

```diff
@@ -96,26 +97,26 @@ message TransactionResult {
 }
 
 message TransactionStreamRequest {
-  uint32 start_layer = 1;
-  uint32 end_layer = 2;
-  repeated bytes id = 3;
-  string address = 4;
-  bool watch = 5;
+  uint32 start_layer = 1; // Starting layer for transaction streaming.
+  uint32 end_layer = 2; // Ending layer for transaction streaming.
+  repeated bytes txid = 3; // Filter: specific transaction IDs.
+  string address = 4; // Filter: address involved in transaction.
+  bool watch = 5; // Continue watching for new transactions.
```

Comment:
I don't understand this. By definition, using a stream means you're subscribing to new txs aren't you? What would happen if this were set to false?

Reply:
Stream is a private service - you can "stream" transactions for specific range of layers. Which will give you all transactions without limit/offset.

Reply:
@lrettig 
> By definition, using a stream means you're subscribing to new txs aren't you?

It's not necessary. A server-streaming response differs from a unary response in that it doesn't return all elements at once but "streams" them one by one. [The GRPC spec](https://grpc.io/docs/what-is-grpc/core-concepts/#server-streaming-rpc) says the server can close the call when it finishes. In our case, when `watch=false`, it ends after streaming all TXs from the DB.

Reply:
Thanks guys, this makes sense